src/Command/DrupalUpdate/CheckPackageInfo.php

```diff
@@ -0,0 +1,210 @@
+<?php
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+use Composer\Semver\Comparator;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+class CheckPackageInfo
+{
+  /**
+   * @var array
+   */
+  public array $infoPackageFiles = [];
+
+  /**
+   * @var array
+   */
+  public array $infoDetailPackages = [];
+
+  /**
+   * @var string[]
+   */
+  public array $packageInfoKey = [
+        'name',
+        'description',
+        'package',
+        'version',
+        'core',
+        'project',
+  ];
+
+  /**
+   * @var array
+   */
+  public array $availablePackageUpdates = [];
+  /**
+   * @var string
+   */
+  private string $drupalRootDirPath;
+  /**
+   * @var string
+   */
+  private string $drupalCoreVersion;
+
+
+  /**
+   * Flag for drupal core update only single time.
+   * Get updates only single time of core for all core modules, themes, profile.
+   * @var bool
+   */
+  public bool $isCoreUpdated = FALSE;
+
+  /**
+   * @var UpdateScriptUtility
+   */
+  private UpdateScriptUtility $updateScriptUtility;
+  /**
+   * @var SymfonyStyle
+   */
+  private SymfonyStyle $io;
+
+  /**
+ * @return UpdateScriptUtility
+ */
+  public function getUpdateScriptUtility(): UpdateScriptUtility {
+    return $this->updateScriptUtility;
+  }
+
+  /**
+   * @param UpdateScriptUtility $updateScriptUtility
+   */
+  public function setUpdateScriptUtility(UpdateScriptUtility $updateScriptUtility): void {
+    $this->updateScriptUtility = $updateScriptUtility;
+  }
+
+  /**
+   * @return mixed
+   */
+  public function getDrupalCoreVersion() {
+    return $this->drupalCoreVersion;
+  }
+
+  /**
+   * @param mixed $drupalCoreVersion
+   */
+  public function setDrupalCoreVersion($drupalCoreVersion): void {
+    $this->drupalCoreVersion = $drupalCoreVersion;
+  }
+
+  /**
+   * @return mixed
+   */
+  public function getDrupalRootDirPath() {
+    return $this->drupalRootDirPath;
+  }
+
+  /**
+   * @param mixed $drupalRootDirPath
+   */
+  public function setDrupalRootDirPath($drupalRootDirPath): void {
+    $this->drupalRootDirPath = $drupalRootDirPath;
+  }
+
+  /**
+   * CheckPackageInfo constructor.
+   * @param InputInterface $input
+   * @param OutputInterface $output
+   */
+  public function __construct(InputInterface $input,
+                              OutputInterface $output) {
+    $this->setUpdateScriptUtility(new UpdateScriptUtility($input, $output));
+    $this->io = new SymfonyStyle($input, $output);
+  }
+
+  /**
+   * @param $filepath
+   * @param $package
+   */
+  public function fileGetInfo($filepath, $package) {
+    set_error_handler(function () {
+        // @todo when multi-dimension array in .info file.
+    });
+    $info_extention_file =  parse_ini_file($filepath, FALSE, INI_SCANNER_RAW);
```

Comment:
@grasmash @balsama parse_ini_file method not working in below cases.
- when multidimensional array exist in .info file ( stylesheet,jquery,library etc.), Below warning throwing 
<img width="1186" alt="Screenshot 2022-06-16 at 6 39 00 PM" src="https://user-images.githubusercontent.com/75003339/174076719-6e1fa09b-a8e3-42d4-b75d-9f47a88b5d16.png">
Could you please suggest if is there any possible solution for this.
or we just handle the warning?

Reply:
It return the required package details that we are using in our code, only the PHP warnings we need to handle.

Reply:
<img width="793" alt="Screenshot 2022-06-16 at 6 46 26 PM" src="https://user-images.githubusercontent.com/75003339/174078209-ddcf5e58-75bf-46c5-b8d4-734b59dc1064.png">